ui/types/utils.d.ts

```diff
@@ -103,11 +103,15 @@ export function setCssVar(
   element?: Element
 ): void;
 
-export class EventBus {
-  on(event: string, callback: Function, ctx?: any): this;
-  once(event: string, callback: Function, ctx?: any): this;
-  emit(event: string, ...args: any[]): this;
-  off(event: string, callback?: Function): this;
+interface Callbacks {
+  [key: string]: (...args: any[]) => void;
+}
+
+export class EventBus<T extends Callbacks> {
+  on<K extends keyof T>(event: K, callback: T[K], ctx?: any): this;
+  once<K extends keyof T>(event: K, callback: T[K], ctx?: any): this;
+  emit<K extends keyof T>(event: K, ...args: Parameters<T[K]>): this;
+  off<K extends keyof T>(event: K, callback: T[K]): this;
```

Comment:
`callback` should be optional as before. When it's not passed, all the listeners for the given event will be removed.

Reply:
@rstoenescu this should be addressed before the release to avoid any annoyance to our users.